addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -491,7 +494,7 @@ const Link = Widget.extend({
      */
     _updateLinkContent($link, linkInfos, { force = false } = {}) {
         if (force || (this._setLinkContent && (linkInfos.content !== this.data.originalText || linkInfos.url !== this.data.url))) {
-            if (linkInfos.content === this.data.originalText) {
+            if (linkInfos.content === this.data.originalText || this.data.isImage) {
```

Comment:
How about:
You get the link label from linkInfos. If the label is empty but there is HTML content, keep the html.

Something like.
```suggestion
            if (linkInfos.content === this.data.originalText || linkInfos.label === "" && this.data.originalHTML) {
```
and make sure you get label from `_getData`.

Reply:
I tried and found that `linkInfos.label` is equal to URL and this happens because if there is no label content then content would be same as URL. So in this case we should probably use` this.$('input[name="label"]').val()`.

Reply:
> if there is no label content then content would be same as URL.

But isn't it `content = this.$('input[name="label"]').val() || url;`, I'm asking you to have `label = this.$('input[name="label"]').val();`

Reply:
Okay but how can I check that?. There is no such property as `lable` in `linkInfos`.

Reply:
Do you want me to create a new property named `label` and pass it through `linkInfos` ?

Reply:
> There is no such property as lable in linkInfos.

You add it in `_getData` and get it from there.

Reply:
Ok but there is one think to be considered. When we hit save button, linkInfos is not coming from `_getData` function. It is coming through callback from `wysiwyg.js` which our `final_data` object. In other words we have to add another property named `label` in our `final_data` object coming from `save` function in `link_dialog.js`.